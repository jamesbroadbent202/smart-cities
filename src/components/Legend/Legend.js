import React from 'react';
import PropTypes from 'prop-types';
import style from './Legend.scss';

const Legend = props => (
  <div className={props.className}>
    {props.title && <div className={style.title}>{props.title}</div>}
    {props.series.map(item => (
      <div
        key={item.name}
        className={style.legendItem}
      >
        <span
          style={{ background: item.color }}
          className={style.legendItemMarker}
        />

        <span className={style.legendItemText}>{item.name}</span>
      </div>
    ))}
  </div>
);

Legend.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  series: PropTypes.arrayOf(PropTypes.shape({
    color: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  })).isRequired,
};

export default Legend;
