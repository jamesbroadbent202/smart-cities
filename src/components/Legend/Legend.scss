@import '~style/variables';

.title {
  @include type-detail-text--bold;

  padding: 4px 24px 0 0;
}

.legendItem {
  margin-right: 24px;
}

.legendItemMarker {
  display: inline-block;
  border-radius: 50%;
  width: 11px;
  height: 11px;
  margin-right: 5px;
  vertical-align: -2px;
}

.legendItemText {
  @include type-detail-text--bold;
}
