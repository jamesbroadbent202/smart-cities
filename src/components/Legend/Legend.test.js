import React from 'react';
import { shallow } from 'enzyme';
import Legend from './Legend';

const defaultProps = {
  series: [
    {
      color: 'red',
      name: 'House'
    },
    {
      color: 'blue',
      name: 'Apartment'
    },
    {
      color: 'grey',
      name: 'Other'
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <Legend {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render the correct number of legend items', () => {
  const component = shallow(
    <Legend {...defaultProps} />
  );

  expect(component.find('.legendItem').length).toBe(defaultProps.series.length);
});

it('applies the passed in color to the background', () => {
  const component = shallow(
    <Legend {...defaultProps} />
  );

  const secondItemStyle = component.find('.legendItemMarker').at(1).prop('style');

  expect(secondItemStyle.background).toBe(defaultProps.series[1].color);
});
