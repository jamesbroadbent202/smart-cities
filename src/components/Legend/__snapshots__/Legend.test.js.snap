// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className={[undefined]}>
  <div className=\\"legendItem\\">
    <span style={{...}} className=\\"legendItemMarker\\" />
    <span className=\\"legendItemText\\">
      House
    </span>
  </div>
  <div className=\\"legendItem\\">
    <span style={{...}} className=\\"legendItemMarker\\" />
    <span className=\\"legendItemText\\">
      Apartment
    </span>
  </div>
  <div className=\\"legendItem\\">
    <span style={{...}} className=\\"legendItemMarker\\" />
    <span className=\\"legendItemText\\">
      Other
    </span>
  </div>
</div>"
`;
